### PR TITLE
Remember the sort option and direction across multiple sessions

### LINKED ISSUE #1327: Sorting always reset on app close
**Describe the bug**
Sorting always reset when app closes/opens

**To Reproduce**
1. OPen UWP File explorer
2. Right click > Sort > Choose any
3. Close UWP File explorer
4. Open UWP File explorer again
5. Right click > sort
6. The sort selected before, is gone

**Expected behavior**
When I choose a specific sort, every time I access that folder, the sorting must be the one I selected.

**Desktop**
 - OS Version: Windows 10 2004 19041.84
 - App version: 0.11.0

### PR DESCRIPTION
**Pull request by Jaiganeshkumaran**
Fixes #1327

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@Jaiganeshkumaran Would it be possible to store the layout setting as well?

File: Files/Enums/SortOption.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace Files.Enums
 {
-    public enum SortOption
+    public enum SortOption : byte
```
Comment (tsvietOK) [inline Files/Enums/SortOption.cs:3]: Is it needed?

File: Files/View Models/ItemViewModel.cs [hunk 2]
```diff
@@ -143,7 +141,7 @@ public static async Task<StorageFileWithPath> GetFileWithPathFromPathAsync(strin
         }
 
         private bool _IsFolderEmptyTextDisplayed;
-
+
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:144]: Fix formatting

File: Files/View Models/ItemViewModel.cs [hunk 3]
```diff
@@ -157,111 +155,80 @@ public bool IsFolderEmptyTextDisplayed
...
-                    NotifyPropertyChanged("IsSortedAscending");
-                    NotifyPropertyChanged("IsSortedDescending");
-                    OrderFiles();
-                }
-            }
-        }
-
+	public void UpdateSortOptionStatus()
+	{
+		NotifyPropertyChanged("IsSortedByName");
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:160]: Fix formatting

File: Files/View Models/ItemViewModel.cs [hunk 4]
```diff
@@ -157,111 +155,80 @@ public bool IsFolderEmptyTextDisplayed
...
+		NotifyPropertyChanged("IsSortedByDate");
+		NotifyPropertyChanged("IsSortedByType");
+		NotifyPropertyChanged("IsSortedBySize");
+	}
+		
+	public void UpdateSortDirectionStatus()
+	{
+		NotifyPropertyChanged("IsSortedAscending");
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:168]: Fix formatting

File: Files/View Models/ItemViewModel.cs [hunk 5]
```diff
@@ -157,111 +155,80 @@ public bool IsFolderEmptyTextDisplayed
...
                 {
-                    DirectorySortOption = SortOption.Name;
+                    AppSettings.DirectorySortOption = SortOption.Name;
                     NotifyPropertyChanged("IsSortedByName");
-                    NotifyPropertyChanged("DirectorySortOption");
                 }
             }
         }
-
+
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:184]: Fix formatting

File: Files/View Models/SettingsViewModel.cs [hunk 6]
```diff
@@ -93,6 +97,61 @@ public DefaultLanguageModel DefaultLanguage
...
+        public SortDirection DirectorySortDirection
+        {
+            get => Get(_directorySortDirection);
+            set
+            {
+                if (value != _directorySortDirection)
+                {
+                    Set(ref _directorySortDirection, value);
+                    localSettings.Values[LocalSettings.SortDirection] =(byte)value;
+		    App.CurrentInstance?.FilesystemViewModel?.UpdateSortDirectionStatus();
```
Comment (tsvietOK) [inline Files/View Models/SettingsViewModel.cs:123]: Fix formatting. Please make sure all other places are formatted as well.

File: Files/View Models/SettingsViewModel.cs [hunk 7]
```diff
@@ -93,6 +97,61 @@ public DefaultLanguageModel DefaultLanguage
...
+            if (localSettings.Values[LocalSettings.SortOption] == null)
+            {
+                _directorySortOption = SortOption.Name;
+                localSettings.Values[LocalSettings.SortOption] = SortOption.Name;
+            }
+            else
+            {
+                _directorySortOption = (SortOption)localSettings.Values[LocalSettings.SortOption];
+            }
+        }
```
Comment (yaichenbaum) [inline Files/View Models/SettingsViewModel.cs:140]: Can you try using a format similar to this for storing the settings?
```
private string _AppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);

public string AppDataPath
{
    get => _AppDataPath;
    set => Set(ref _AppDataPath, value);
}
```

Comment (Jaiganeshkumaran) [reply Files/View Models/SettingsViewModel.cs:140]: This is for detecting the sort option from AppData.

Comment (yaichenbaum) [reply Files/View Models/SettingsViewModel.cs:140]: This method does not require you to "detect" the setting as it does it for you.

File: Files/Enums/SortOption.cs (hunk 1 above)
Comment (Jaiganeshkumaran) [reply Files/Enums/SortOption.cs:3]: Just to save memory

File: Files/View Models/SettingsViewModel.cs (hunk 7 above)
Comment (Jaiganeshkumaran) [reply Files/View Models/SettingsViewModel.cs:140]: LocalSettings doesn't support Enums.

Comment (Jaiganeshkumaran) [reply Files/View Models/SettingsViewModel.cs]: Done

**Comment by Jaiganeshkumaran** [comment]
> There is a bug, when trying to open app first time after this patch.

This issue has been fixed.

**Comment by Jaiganeshkumaran** [comment]
> @Jaiganeshkumaran Would it be possible to store the layout setting as well?

Isn't the layout mode stored already?

**Pull request merged**